extensions/default/src/DataSourceConfigurationAPI/GoogleCloudDataSourceConfigurationAPI.ts

```diff
@@ -0,0 +1,192 @@
+import { ExtensionManager, Types } from '@ohif/core';
+
+/**
+ * The various Google Cloud Healthcare path item types.
+ */
+enum ItemType {
+  projects = 0,
+  locations = 1,
+  datasets = 2,
+  dicomStores = 3,
+}
+
+interface NamedItem {
+  name: string;
+}
+interface Project extends NamedItem {
+  projectId: string;
+}
+
+/**
+ * Fetches an array of items the specified item type.
+ * @param urlStr the fetch url
+ * @param fetchItemType the type to fetch
+ * @param fetchOptions the header options for the fetch (e.g. authorization header)
+ * @param fetchSearchParams any search query params; currently only used for paging results
+ * @returns an array of items of the specified type
+ */
+async function _doFetch(
```

Comment:
consider refactoring _doFetch outside

Reply:
Actually _doFetch is closely tied to the Google API, that instead I will make it a private static method inside of the GoogleCloudDataSourceConfigurationAPI class.